### PR TITLE
Add label command with colorized output

### PR DESCRIPTION
**Pull request by darrenwee**
Addresses issue #1280 and based off #1545.

Usage: `hub issue labels` in a repository prints a list of labels with their corresponding colors.

This works okay for repos with a small list of labels (such as this one) but is not as elegant for a repository with a lot of labels. I propose either printing them in lines grouped by their color or paging the output.

### CODE REVIEW

File: commands/issue.go [hunk 1]
```diff
@@ -428,3 +437,55 @@ text is the title and the rest is the description.`, project))
...
+	if args.Noop {
+		ui.Printf("Would request list of labels for %s\n", project)
+		return
+	}
+
+	labels, err := gh.FetchLabels(project)
+	utils.Check(err)
+
+	for _, label := range labels {
+		// TODO add a colorize flag for the labels command
```
Comment (mislav) [inline commands/issue.go]: Would this be enough?

```go
colorize := ui.IsTerminal(os.Stdout)
```

File: commands/issue.go [hunk 2]
```diff
@@ -428,3 +437,55 @@ text is the title and the rest is the description.`, project))
...
+		utils.Check(err)
+	}
+
+	placeholders := map[string]string{
+		"l": label.Name,
+		"c": fmt.Sprintf("\033[38;5;%d;48;2;%d;%d;%dm %s \033[m",
+			getSuitableTextColor(color), color.Red, color.Green, color.Blue, label.Name),
+	}
+
+	return ui.Expand(format, placeholders, colorize)
```
Comment (mislav) [inline commands/issue.go]: I think `ui.Expand` with placeholders might have been an overkill here. If the user doesn't have the option to pass the `--format` string (like for `hub issue` or `hub pr list`), then we could get away with simply:

```go
if colorize {
  return fmt.Sprintf("output with color", ...)
} else {
  return fmt.Sprintf("%s\n", label.Name)
}
```

File: commands/issue.go [hunk 3]
```diff
@@ -428,3 +437,55 @@ text is the title and the rest is the description.`, project))
...
+	}
+
+	color, err := utils.NewColor(label.Color)
+	if err != nil {
+		utils.Check(err)
+	}
+
+	placeholders := map[string]string{
+		"l": label.Name,
+		"c": fmt.Sprintf("\033[38;5;%d;48;2;%d;%d;%dm %s \033[m",
```
Comment (mislav) [inline commands/issue.go]: Can the logic to construct the ANSI escape sequence for a specific color be shared between this command and listing issues?

File: commands/issue.go [hunk 4]
```diff
@@ -428,3 +437,55 @@ text is the title and the rest is the description.`, project))
...
+
+	labels, err := gh.FetchLabels(project)
+	utils.Check(err)
+
+	for _, label := range labels {
+		// TODO add a colorize flag for the labels command
+		ui.Printf(formatLabel(label, true))
+	}
+
+	args.NoForward()
```
Comment (mislav) [inline commands/issue.go]: `NoForward` needs to be above `if args.Noop()` because of the potential early `return`

File: commands/issue.go (hunk 1 above)
Comment (darrenwee) [reply commands/issue.go]: I was thinking along the lines of a command line argument to control whether to print in plain text or a colorized string, like `hub issue labels --no-color`. I could add that in plus the `IsTerminal` check?

**Comment by mislav** [comment]
Thank you! I have swapped the `--no-color` flag in favor of `--color` (explicit opt-in) because I agree with you that the default colored output isn't very useful when labels stand on their own. I imagine that people will be mostly using this command for scripting anyway, where colors are irrelevant.

**Pull request merged**